block/src/Pos/Block/Network/Logic.hs

```diff
@@ -491,8 +493,11 @@ applyWithoutRollback
     -> OldestFirst NE Block
     -> m ()
 applyWithoutRollback enqueue blocks = do
-    logInfo $ sformat ("Trying to apply blocks w/o rollback: "%listJson) $
-        fmap (view blockHeader) blocks
+    let headers = getOldestFirst $ map (view blockHeader) blocks
+    logInfo $ sformat ("Trying to apply blocks w/o rollback. First headers: "%listJson
+                      %"\nLast headers: "%listJson)
+        (NE.take 3 headers)
+        (takeLast 3 headers)
```

Comment:
It's a bit more complicated. If there are less than 6 headers, some of them will be printed more than once which is not what we want. There's the same problem below.